src/main/webapp/management-console.js

```diff
@@ -304,38 +305,60 @@ angular.module('managementConsole', [
             .when('/', '/login')
             .when('', '/login')
             .otherwise('/error404');
+    }])
+
+  .config(['localStorageServiceProvider', function(localStorageServiceProvider) {
+    localStorageServiceProvider.setPrefix('infinispan-management-console');
   }])
+
 /**
  * Safe apply method. It should be used when normal $scope.$apply happens to execute
  * during digest cycle which causes an error.
  * Use it just like normal apply: $scope.safeApply(myFunc).
  */
   .run(['$rootScope', '$timeout', function ($rootScope, $timeout) {
-      $rootScope.safeApply = function (f) {
-        var scope = this;
-        $timeout(function () {
-          scope.$apply(f);
-        });
-      };
-      $rootScope.page = {htmlClass: ''};
-    }]).run(['$rootScope', '$location', 'modelController', function ($rootScope, $location, modelController) {
+    $rootScope.safeApply = function (f) {
+      var scope = this;
+      $timeout(function () {
+        scope.$apply(f);
+      });
+    };
+    $rootScope.page = {htmlClass: ''};
+  }])
 
-      $rootScope.$on('$locationChangeStart', function () {
+  .run([
+    '$rootScope', 'modelController', '$urlRouter', '$state',
+    function ($rootScope, modelController, $urlRouter, $state) {
+      $rootScope.$on('$stateChangeStart', function (event, toState) {
         // redirect to login page if not logged in
-        if ($location.path() !== '/login' && !modelController.isAuthenticated()) {
-          $location.path('/login');
+        if (toState.name !== 'login' && !modelController.isAuthenticated()) {
+          event.preventDefault();
+          modelController.login().then(function () {
+            // TODO: Do I have to do this refreshing here? Why?
```

Comment:
Hmm, I think we need call here because we do not use login controller to login. Admittedly I made things a bit more convoluted than they need to be @Martinsos